model/src/main/kotlin/OrtResult.kt

```diff
@@ -446,11 +446,14 @@ data class OrtResult(
      * [omitResolved] and remove violations below the [minSeverity].
      */
     @JsonIgnore
-    fun getRuleViolations(omitResolved: Boolean = false, minSeverity: Severity? = null): List<RuleViolation> {
+    fun getRuleViolations(
+        omitResolved: Boolean = false,
+        minSeverity: Severity = Severity.entries.min()
+    ): List<RuleViolation> {
         val allViolations = evaluator?.violations.orEmpty()
 
         val severeViolations = when (minSeverity) {
-            null -> allViolations
+            Severity.entries.min() -> allViolations
```

Comment:
With nullable `minSeverity` it was natural to only filter if `minSeverity != null`, but now this optimization seems a bit premature and referring to `Severity.entries.min()` twice is not so nice, IMO. So maybe just drop this optimization and always filter? When doing so, this should of course also be mentioned in the commit message as a conscious decision.

Reply:
I believe the same applies to the performance optimization below for `omitResolved`. So, I did that in a following separate commit.